api/tests/unit/application/assessments/assessment-controller_test.js

```diff
@@ -5,79 +5,84 @@ const assessmentController = require('../../../../lib/application/assessments/as
+    let codeStub;
+    let replyStub;
 
-      // then
-      return promise.then(() => {
-        expect(assessmentService.findByFilters).to.have.been.called;
-        expect(assessmentService.findByFilters).to.have.been.calledWith({ courseId: 'courseId' });
-      });
+    const assessments = [{ id: 1 }, { id: 2 }];
+    const assessmentsInJSONAPI = [
+      {
+        id: 1,
+        type: 'assessments',
+        attributes: { pixScore: 12 }
+      }, {
+        id: 1,
+        type: 'assessments',
+        attributes: { pixScore: 12 }
+      }];
+
+    beforeEach(() => {
+      sandbox = sinon.sandbox.create();
+
+      codeStub = sinon.stub();
+      replyStub = sinon.stub().returns({ code: codeStub });
+      sandbox.stub(assessmentService, 'findByFilters').resolves();
+      sandbox.stub(assessmentSerializer, 'serializeArray').resolves();
+    });
+
+    afterEach(() => {
+      sandbox.restore();
     });
 
-    it('should serialize assessment to JSON API', function() {
-      // given
-      assessmentService.findByFilters.resolves(assessments);
+    describe('GET assessments with filters', function () {
 
-      // when
-      const promise = assessmentController.findByFilters(request, replyStub);
+      const request = { query: { 'filter[courseId]': 'courseId' } };
 
-      // then
-      return promise.then(() => {
-        expect(assessmentSerializer.serializeArray).to.have.been.called;
-        expect(assessmentSerializer.serializeArray).to.have.been.calledWith(assessments);
+      it('should call assessment service with query filters', function () {
+        // when
+        const promise = assessmentController.findByFilters(request, replyStub);
+
+        // then
+        return promise.then(() => {
+          expect(assessmentService.findByFilters).to.have.been.called;
+          expect(assessmentService.findByFilters).to.have.been.calledWith({ courseId: 'courseId' });
+        });
       });
-    });
 
-    it('should reply the serialized assessments', function() {
-      // given
-      assessmentSerializer.serializeArray.returns(assessmentsInJSONAPI);
+      it('should serialize assessment to JSON API', function () {
+        // given
+        assessmentService.findByFilters.resolves(assessments);
 
-      // when
-      const promise = assessmentController.findByFilters(request, replyStub);
+        // when
+        const promise = assessmentController.findByFilters(request, replyStub);
 
-      // then
-      return promise.then(() => {
-        expect(replyStub).to.have.been.called;
-        expect(replyStub).to.have.been.calledWith(assessmentsInJSONAPI);
+        // then
+        return promise.then(() => {
+          expect(assessmentSerializer.serializeArray).to.have.been.called;
+          expect(assessmentSerializer.serializeArray).to.have.been.calledWith(assessments);
+        });
       });
+
+      it('should reply the serialized assessments', function () {
```

Comment:
Ton test dit `should REPLY` mais tu vérifie seulement des appels, il faudrais peut-être vérifier (`deepEqual`) que l'objet retourné correspond à se que tu veux.